### ISSUE TITLE
Discussion of whether to use `~/.emacs.d/private/snippets`

### ISSUE THREAD

**Issue by CestDiego** [syl20bnr/spacemacs#1607]
I would like to discuss whether it's better to use the private dir to store something so personal as custom snippets, most of the time one would like to VC their snippets, (if they are not auto generated like the ones @tuhdo uses)

So I'd like to suggest that we use something like a layer variable for `custom-private-snippets that are stored anywhere we might find cool to put it. Like maybe inside the   
`dotspacemacs-configuration-layer-path`.

Just putting this in the `dotspacemacs/config` doesn't cut it

```elisp
  (setq my-yas-dir (concat (car dotspacemacs-configuration-layer-path)
                           "snippets/"))
  (setq yas-snippet-dirs
        (append (list my-yas-dir)
                (when (boundp 'yas-snippet-dirs)
                  yas-snippet-dirs)))
```

Because even though `dotspacemacs/config` gets evaluated after all layers have been loaded, the yasnippet stuff is lazily loaded so then `~/.emacs.d/private/snippets/` gets prepended to the same variable.

**Comment by syl20bnr** [comment]
Good idea, this would be an `auto-completion` layer variable so we can name it `auto-completion-private-snippet-directory`.

**Comment by CestDiego** [comment]
Yes! This would be very awesome indeed.

**Comment by herbert-jones-mediafire** [comment]
This is overridding my own private snippets layer now.  As yasnippet selects the first directory in ```yas-snippet-dirs``` to save new snippets to, all my new snippets end up in private instead of the directory I have set in my config.  My snippets are still loaded, as their directory is second.  This change breaks setting ```yas-snippet-dirs``` in .spacemacs or in a layer for anyone that expects the setting to be honored.

**Comment by justbur** [comment]
@TheBB This is implemented in #1861
